### PR TITLE
fix: Cache access token validation

### PR DESCRIPTION
**Pull request by JigarJoshi**
- Added one switch that will control if the auth middleware should be invoked or not.
- cache access token for better performance.
- validate expiration after reading from cache.
- TODO: handle key rotation

### CODE REVIEW

File: server/config/options.go [hunk 1]
```diff
@@ -118,6 +121,7 @@ var DefaultConfig = Config{
 		FDBHardDrop: false,
 	},
 	Auth: AuthConfig{
+		EnableAuthCheck:  false,
```
Comment (efirs) [inline server/config/options.go:124]: Can this be just Enabled?

Comment (efirs) [reply server/config/options.go:124]: to be consistent with other packages configs.

File: server/middleware/auth.go [hunk 2]
```diff
@@ -138,12 +139,26 @@ func AuthFunction(ctx context.Context, jwtValidator *validator.Validator, config
...
+		validatedToken, err = jwtValidator.ValidateToken(ctx, tkn)
+		if err != nil {
+			log.Warn().Err(err).Msg("Failed to validate access token")
+			return ctx, api.Errorf(api.Code_UNAUTHENTICATED, "Failed to validate access token")
+		}
+		cache.Add(tkn, validatedToken)
+	} else {
+		// validate expiration
+		validatedClaims, ok := validatedToken.(*validator.ValidatedClaims)
+		if ok {
```
Comment (efirs) [inline server/middleware/auth.go:153]: This check can be moved inside below `if`.

Comment (efirs) [reply server/middleware/auth.go:153]: so the else branch is not needed here.

File: server/config/options.go (hunk 1 above)
Comment (JigarJoshi) [reply server/config/options.go]: updated. thanks

File: server/middleware/auth.go (hunk 2 above)
Comment (JigarJoshi) [reply server/middleware/auth.go]: updated. thanks

**Review by efirs — APPROVED** [review]

**Pull request merged**